### PR TITLE
fast json parser didn't like empty byte array

### PR DESCRIPTION
**Pull request by yingjieMiao**
@eishay  @raymondng 

we have many errors from scraper that were caused by this. This doesn't fix the whole problem though, in the shoebox client, we will have JsNull.as[SomeType]. We could handle that gracefully elsewhere.

But as a parser, FastJsonParser should handle empty byte array correctly.

### CODE REVIEW

**Comment by eishay** [comment]
nice!

File: kifi-backend/modules/common/app/com/keepit/common/net/FastJsonParser.scala [hunk 1]
```diff
@@ -50,7 +50,8 @@ class FastJsonParser() {
    */
   private def fastParse(bytes: Array[Byte]) = if (bytes.size <= 4) { //4 == "null".getBytes(UTF8).size
     val smallString = new String(bytes, UTF8)
-    if (smallString == FastJsonParser.emptyObjectString) FastJsonParser.emptyObject
+    if (bytes.isEmpty) FastJsonParser.nullVal
```
Comment (raymondng) [inline kifi-backend/modules/common/app/com/keepit/common/net/FastJsonParser.scala:53]: how about checking this before we construct smallString or do so in the if-expr?

**Comment by raymondng** [comment]
agreed -- we should handle empty array correctly. :+1:

File: kifi-backend/modules/common/app/com/keepit/common/net/FastJsonParser.scala (hunk 1 above)
Comment (yingjieMiao) [inline kifi-backend/modules/common/app/com/keepit/common/net/FastJsonParser.scala:53]: sure. i can change that.

Comment (raymondng) [inline kifi-backend/modules/common/app/com/keepit/common/net/FastJsonParser.scala:53]: Thanks!

**Pull request merged**